clients/unshaded/src/main/java/tachyon/client/file/FileSystem.java

```diff
@@ -0,0 +1,375 @@
+   *
+   * @param path the path of the directory to create in Tachyon space
+   * @return the {@link TachyonURI} referencing the newly created directory
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected exception is thrown
+   */
+  TachyonURI createDirectory(TachyonURI path)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Creates a directory.
+   *
+   * @param path the path of the directory to create in Tachyon space
+   * @param options options to associate with this operation
+   * @return the {@link TachyonURI} referencing the newly created directory
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  TachyonURI createDirectory(TachyonURI path, CreateDirectoryOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #createFile(TachyonURI, CreateFileOptions)} with default options.
+   *
+   * @param path the path of the file to create in Tachyon space
+   * @return a {@link FileOutStream} which will write data to the newly created file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  FileOutStream createFile(TachyonURI path)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Creates a file.
+   *
+   * @param path the path of the file to create in Tachyon space
+   * @param options options to associate with this operation
+   * @return a {@link FileOutStream} which will write data to the newly created file
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileAlreadyExistsException if there is already a file at the given path
+   * @throws InvalidPathException if the path is invalid
+   * @throws TachyonException if an unexpected Tachyon exception is thrown
+   */
+  FileOutStream createFile(TachyonURI path, CreateFileOptions options)
+      throws FileAlreadyExistsException, InvalidPathException, IOException, TachyonException;
+
+  /**
+   * Convenience method for {@link #delete(TachyonURI, DeleteOptions)} with default options.
+   *
+   * @param path the path to delete in Tachyon space
+   * @throws IOException if a non-Tachyon exception occurs
+   * @throws FileDoesNotExistException if the given path does not exist
```

Comment:
Could there be an option which do not treat FileDoesNotExistException as error? In some sense, `delete` a non-existent path "succeeds". 

This is useful when a user just want to make sure a file/directory doesn't exist, if exists, delete it. This requirement can be satisfied using `exists`, but then the code looks like below, which is too verbose.

```
  if (exists(path)) {
    try {
      delete(path);
    } catch (FileDoesNotExistException e) {
       // what's the appropriate handling here?
    }
  }
```